File: 2025-09/2025-09-30/phoneNumberFormatter.js
function formatNumber(number) {
  const { countryCode, areaCode, localNumber1, localNumber2 } = 
    /(?<countryCode>\d)(?<areaCode>\d{3})(?<localNumber1>\d{3})(?<localNumber2>\d{4})/
      .exec(number)
      .groups;

  return `+${countryCode} (${areaCode}) ${localNumber1}-${localNumber2}`;
}

console.log(formatNumber("05552340182"));
console.log(formatNumber("15554354792"));

File: 2025-09/2025-09-30/phone_number_formatter.py
def format_number(number):
    import re
    groups_dict = re \
        .match(r"(?P<countryCode>\d)(?P<areaCode>\d{3})(?P<localNumber1>\d{3})(?P<localNumber2>\d{4})", number) \
        .groupdict()

    return f"+{groups_dict['countryCode']} ({groups_dict['areaCode']}) {groups_dict['localNumber1']}-{groups_dict['localNumber2']}"

print(format_number("05552340182"))
print(format_number("15554354792"))